src/sema/variables.rs

```diff
@@ -250,6 +252,8 @@ pub fn var_decl(
         ty: ty.clone(),
         constant: is_constant,
         initializer,
+        read: false,
+        assigned: initialized,
```

Comment:
`initializer.is_some()` ?

Reply:
I changed `initializer` to `initializer: initializer.clone()` as well.